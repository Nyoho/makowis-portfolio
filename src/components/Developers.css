:local .developers > :not(:first-child) {
    margin: 0 0 0 10px;
}
