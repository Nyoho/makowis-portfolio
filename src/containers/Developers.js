/* eslint-env browser */
// @flow
import { connect } from 'react-redux';

import type { Dispatch } from 'redux';

import Component from '../components/Developers';
import { developers } from '../models/Person';
import { changePerson } from '../reducers/person';

const mapStateToProps = () => ({
  people: developers,
});

// noinspection JSUnusedGlobalSymbols
const mapDispatchToProps = (dispatch: Dispatch) => ({
  // なぜか IDEA に Unusedっていわれる。
  onPersonClick: (person) => () => {
    // TODO react-router を導入すれば削除できるので、直接扱う
    window.history.pushState(null, person.name, `/${person.id}`);
    dispatch(changePerson(person));
  },
});

const Developers = connect(mapStateToProps, mapDispatchToProps)(Component);

export default Developers;
