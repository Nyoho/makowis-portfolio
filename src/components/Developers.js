// @flow
import React from 'react';
import Avatar from './Avatar';
import { developers as className } from './Developers.css';
import type { Person } from '../models/Person';

const AvatarForPerson = ({ avatar, name, ...rest }: Person) => (
  <Avatar {...{ ...rest, src: avatar, alt: name }} />
);

type Props = {
  people: Person[],
  onPersonClick: (Person) => (SyntheticTouchEvent<any>) => void,
};

const Developers = ({ people, onPersonClick }: Props) => (
  <div {...{ className }}>
    {people.map((person) => (
      <AvatarForPerson
        {...person}
        key={person.id}
        onClick={onPersonClick(person)}
        width={40}
        height={40}
      />
    ))}
  </div>
);

export default Developers;
