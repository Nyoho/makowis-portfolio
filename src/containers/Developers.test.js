/* eslint-env jest */
// @flow
import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Developers from './Developers';

const reducer = () => ({});

const store = createStore(reducer);

it('renders Person Developers', () => {
  const tree = renderer
    .create(
      <Provider {...{ store }}>
        <Developers />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
