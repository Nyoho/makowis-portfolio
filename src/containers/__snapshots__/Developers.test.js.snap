// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Person Developers 1`] = `
<div
  className={undefined}
>
  <img
    alt="Makoto Henmi"
    className={undefined}
    facebook="makoto.henmi"
    github="makowis"
    height={40}
    id="mako_wis"
    nickname="mako_wis"
    onClick={[Function]}
    speakerdeck="makowis"
    src="avatar.bmp"
    twitter="mako_wis"
    width={40}
  />
  <img
    alt="Tomohiko Himura"
    className={undefined}
    facebook="eielh"
    github="eiel"
    height={40}
    id="eiel"
    nickname="eiel"
    onClick={[Function]}
    speakerdeck="eiel"
    src="https://pbs.twimg.com/profile_images/2754435592/6fef2556bc06febd5c6ab4c24ae08282_400x400.jpeg"
    twitter="eielh"
    width={40}
  />
</div>
`;
